### PR TITLE
Fix for Issue #1013

### LINKED ISSUE #1013: Allow user to sort issue
Sort issues on the issues explorer table.

### PR DESCRIPTION
**Pull request by webbdays**
# Pull Request Template

> supposed to Closes #1013 (not yet)

## Description
To allow user to sort the columns in a table

## Purpose
To allow user to sort the columns in a table

## Solution
Using **TableSortLabel** component from **MUI**, Got  to know about it after seeing the PR's related to this issue.

## Screenshots
**None**

## Checklist
- [ ] Code has been tested and works as intended - tests for my code changes not written yet, but tests on previous codebase are failing
- [ ] Properly documented code - not yet
- [ ] Follows coding standards and guidelines - not sure
- [ ] All tests pass - tests on previous codebase are failing
- [ ] No conflicts with current codebase - not sure, tests for my code changes not written yet to confirm this.
- [ ] Code has been reviewed by at least one other team member - not yet

## Related Issues
None

## Additional Notes
None

## Contributor Guidelines
Fork the repo and create your branch from master. -  done
If you've added code that should be tested, add tests. - not yet
If you've changed APIs, update the documentation. - NA
Ensure the test suite passes. - tests on previous codebase are failing
Make sure your code lints. - linting on previous codebase are failing
Issue that pull request! - going to do it
Check if the build is green before we start to review - waiting

Thank you.

### CODE REVIEW

**Comment by webbdays** [comment]
The problem with tests for previous code base is at Transfer.update()
its not returning datavalues.

**Comment by webbdays** [comment]
I think hooks need to be setup like it was done for Task to store update history.

**Comment by webbdays** [comment]
tests pass for node v17.3.0

File: frontend/src/components/task/task-table.js [hunk 1]
```diff
@@ -186,38 +214,58 @@ class CustomPaginationActionsTable extends React.Component {
...
+                  this.setState({sortByField : fieldId});
+                  return sortHandler(fieldId, this.state.sortDirection ==="asc" ? "desc" : "asc" )
+                }
+              }
+            >
+              <FormattedMessage id={fieldId} defineMessage={defineMessage}/>
+            </TableSortLabel>
+      )
+    }
+    const  TableHeadNew = ()=> {
```
Comment (alexanmtz) [inline frontend/src/components/task/task-table.js:234]:
```suggestion
    const  TableHead = ()=> {
```

File: frontend/src/components/task/task-table.js [hunk 2]
```diff
@@ -186,38 +214,58 @@ class CustomPaginationActionsTable extends React.Component {
...
-                    </TableCell>
-                    <TableCell>
-                      <FormattedMessage id='task.table.head.labels' defaultMessage='Labels' />
-                    </TableCell>
-                    <TableCell>
-                      <FormattedMessage id='task.table.head.createdAt' defaultMessage='Created' />
-                    </TableCell>
-                  </TableRow>
-                </TableHead>
+              <TableHeadNew />
```
Comment (alexanmtz) [inline frontend/src/components/task/task-table.js:266]:
```suggestion
              <TableHead />
```

**Review by alexanmtz — APPROVED** [review]

**Comment by alexanmtz** [comment]
Great @webbdays with the solution, I just changed the name of the component to make it more concise, but great job and well done with the build fix! 🎉

**Comment by webbdays** [comment]
Thank you.

**Comment by webbdays** [comment]
Thanks to PR's #1068 #1069 .

**Comment by webbdays** [comment]
But tests for new functionality in frontend are not covered. And also code docs.

**Comment by alexanmtz** [comment]
@webbdays I will have to revert this PR, still an issue on the logged page:
pageshot of 'Gitpay - Bounties for git issues solved' @ 2024-02-20-1542'57

Please check it and create a new PR with your fixes.

**Comment by webbdays** [comment]
ok.

**Pull request merged**